tests/compose/core/05_connectivity.py

```diff
@@ -0,0 +1,106 @@
+        conn.noop()
+    print('OK')
+    print(f'Authenticating to imap://{username}:{password}@{server}:143/')
+    try:
+        with imaplib.IMAP4(server) as conn:
+            conn.login(username, password)
+        print(f'Authenticating to imap://{username}:{password}@{server}:143/ worked without STARTTLS!')
+        os.exit(102)
+    except imaplib.IMAP4.error:
+        print('NOK - expected')
+
+def test_pop3(server, username, password):
+    print(f'Authenticating to pop3s://{username}:{password}@{server}:995/')
+    conn = poplib.POP3_SSL(server)
+    conn.capa()
+    conn.user(username)
+    conn.pass_(password)
+    conn.close()
+    print('OK')
+    print(f'Authenticating to pop3s://{username}:{password}@{server}:110/')
+    conn = poplib.POP3(server)
+    conn.stls()
+    conn.capa()
+    conn.user(username)
+    conn.pass_(password)
+    conn.close()
+    print('OK')
+    print(f'Authenticating to pop3://{username}:{password}@{server}:110/')
+    try:
+        conn = poplib.POP3(server)
+        conn.capa()
+        conn.user(username)
+        conn.pass_(password)
+        conn.close()
+        print(f'Authenticating to pop3://{username}:{password}@{server}:110/ worked without STARTTLS!')
+        os.exit(103)
+    except poplib.error_proto:
+        print('NOK - expected')
+
+def test_SMTP(server, username, password):
+    print(f'Authenticating to smtps://{username}:{password}@{server}:465/')
+    with smtplib.SMTP_SSL(server) as conn:
+        conn.ehlo()
+        conn.login(username, password)
+        print('OK')
+    print(f'Authenticating to smtps://{username}:{password}@{server}:587/')
+    with smtplib.SMTP(server, 587) as conn:
+        conn.ehlo()
+        conn.starttls()
+        conn.ehlo()
+        conn.login(username, password)
+        print('OK')
+    try:
+        print(f'Authenticating to smtp://{username}:{password}@{server}:587/')
+        with smtplib.SMTP(server, 587) as conn:
+            conn.ehlo()
+            conn.login(username, password)
+            print(f'Authenticating to smtp://{username}:{password}@{server}:587/ worked!')
+    except smtplib.SMTPNotSupportedError:
+        print('NOK - expected')
+    #port 25 should fail
+    try:
+        print(f'Authenticating to smtps://{username}:{password}@{server}:25/')
+        with smtplib.SMTP(server) as conn:
+            conn.ehlo()
+            conn.starttls()
+            conn.ehlo()
+            conn.login(username, password)
+            print(f'Authenticating to smtps://{username}:{password}@{server}:25/ worked!')
+    except smtplib.SMTPNotSupportedError:
+        print('NOK - expected')
+    try:
+        print(f'Authenticating to smtp://{username}:{password}@{server}:25/')
+        with smtplib.SMTP(server) as conn:
+            conn.ehlo()
+            conn.login(username, password)
+            print(f'Authenticating to smtp://{username}:{password}@{server}:25/ worked without STARTTLS!')
+            os.exit(104)
```

Comment:
```suggestion
            os.exit(106)
```
due to the above changes this number must be bumped to make it unique.